src/main/java/eu/europeana/metis/sandbox/controller/PatternAnalysisController.java

```diff
@@ -71,13 +74,19 @@ public PatternAnalysisController(PatternAnalysisService<Step, ExecutionPoint> pa
       @ApiResponse(code = 404, message = "Dataset not found")
   })
   @GetMapping(value = "{id}/get-dataset-pattern-analysis", produces = APPLICATION_JSON_VALUE)
-  public DatasetProblemPatternAnalysisView<Step> getDatasetPatternAnalysis(
-      @ApiParam(value = "id of the dataset", required = true) @PathVariable("id") String datasetId,
-      @ApiParam(value = "timestamp of when the step was executed", required = true) @RequestParam LocalDateTime executionTimestamp) {
-    return new DatasetProblemPatternAnalysisView<>(
-        patternAnalysisService.getDatasetPatternAnalysis(datasetId, Step.VALIDATE_INTERNAL, executionTimestamp).orElse(
-            new DatasetProblemPatternAnalysis<>("0", null, null, new ArrayList<>())));
-
+  public ResponseEntity<DatasetProblemPatternAnalysisView<Step>> getDatasetPatternAnalysis(
+      @ApiParam(value = "id of the dataset", required = true) @PathVariable("id") String datasetId) {
+    Optional<ExecutionPoint> executionPointOptional = executionPointService.getExecutionPoint(datasetId,
+        Step.VALIDATE_INTERNAL.toString());
+    if (executionPointOptional.isPresent()) {
+      return new ResponseEntity<>(new DatasetProblemPatternAnalysisView<>(
+          patternAnalysisService.getDatasetPatternAnalysis(datasetId, Step.VALIDATE_INTERNAL,
+              executionPointOptional.get().getExecutionTimestamp()).orElse(
```

Comment:
Why do you have this `if...else` block? The `.orElse()` already does this check for you and returns the actual value if is present. What is inside the `orElse()` is the value to return in case **it is not present**.
This `if...else` block is not necessary and it's just repeating behavior `.orElse()` already does for you.
```suggestion
             return new ResponseEntity<>(new DatasetProblemPatternAnalysisView<>(
          patternAnalysisService.getDatasetPatternAnalysis(datasetId, Step.VALIDATE_INTERNAL,
              executionPointOptional.get().getExecutionTimestamp()).orElse(new DatasetProblemPatternAnalysis<>("0", null, null, new ArrayList<>()))), HttpStatus.NOT_FOUND);
```

Reply:
I will have a look on rewrite it to return the http status code also when the dataset id is not found. That is the main reason of the ```if...else``` block

Reply:
Thanks for the check, actually there was a missing return value in the orElse part so I rewrote this part of the code and now covers two different conditions.
1. If the dataset exists. will proceed to do the analysis else return a not found of the dataset.
2. If the pattern analysis exists. will proceed to process it else return a not found of the analysis.